### ISSUE TITLE
loss became nan after first optimizer step

### ISSUE THREAD

**Issue by MasterEndless** [Lightning-AI/lit-gpt#291]
I try to fine-tune falcon 7b using adapter.py on 8 V100 with 32GB, my configuration is as following:
devices = 8
override_max_seq_length = 1024
micro_batch_size = 1
And I am using 16-true.

The loss will began to become NAN after the first optimizer step, any ideas on how to solve this?

Estimated TFLOPs: 768.16
Measured TFLOPs: 739.53
iter 0 step 0: loss 0.5254, iter time: 5439.32ms
iter 1 step 0: loss 1.5451, iter time: 1806.42ms
iter 2 step 0: loss 2.3252, iter time: 1831.11ms
iter 3 step 0: loss 1.1392, iter time: 1037.10ms
iter 4 step 0: loss 1.3027, iter time: 2145.38ms
iter 5 step 0: loss 3.6702, iter time: 1352.48ms
iter 6 step 0: loss 1.1289, iter time: 1897.51ms
iter 7 step 1: loss 1.1201, iter time: 1819.72ms (optimizer.step)
iter 8 step 1: loss nan, iter time: 1957.89ms
iter 9 step 1: loss nan, iter time: 1439.00ms
iter 10 step 1: loss nan, iter time: 2076.89ms
iter 11 step 1: loss nan, iter time: 1359.93ms
iter 12 step 1: loss nan, iter time: 825.32ms
iter 13 step 1: loss nan, iter time: 814.55ms
iter 14 step 1: loss nan, iter time: 2005.71ms

**Comment by Andrei-Aksionov** [comment]
Perhaps it's because you are using `16-true`. You can find an explanation in this [blogpost](https://lightning.ai/pages/community/tutorial/pytorch-memory-vit-llm/) (section 3).
It looks like `V100` supports `16-mixed`:
https://github.com/Lightning-AI/lit-gpt/blob/bdad14fe078f6c08253fe29026e65aa0f72612d1/lit_gpt/speed_monitor.py#L50-L52
so you should use it instead.

**Comment by MasterEndless** [comment]
@Andrei-Aksionov Thanks for your reply! But when I switch to 16-mixed, the program will OOM even with a very small max_seq_len

**Comment by carmocca** [comment]
This is expected because weights are kept in float32 precision with `16-mixed`. And weights are very large in "Large" language models
